migrations/20170413103541_shopping_list.js

```diff
@@ -0,0 +1,11 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('shopping_list', function (table) {
+    table.increments('id').primary()

```

Comment:
I'm unclear on the relationship here at the moment it looks like a shopping list has many flats which doesn't seem correct.

I expect you want a flat has many shopping lists in which case the foreign key should live on the shopping_list table